### DISCUSSION TITLE
How can I change the current affiliate link suffix word to id or ref?

### DISCUSSION THREAD

**Discussion by immonsterx** [opencart/opencart#14098] [Q&A]
How can I change the current affiliate link suffix word to id or ref?
For example:
123.com/product?tracking=ABCDE is changed to
123.com/product?id=ABCDE

I changed the '&tracking=' in the /catalog/controller/account/tracking.php file to '&id='. The front-end link is indeed changed, but when tested, this affiliate link has no effect

`foreach ($results as $result) {
$json[] = array(
'name' => strip_tags(html_entity_decode($result['name'], ENT_QUOTES, 'UTF-8')),
'link' => str_replace('&amp;', '&', $this->url->link('product/product', 'product_id='. $result['product_id']. '&id='. $tracking))
);
}
`

Any answer to this would be greatly appreciated! Thanks!

**Comment by mavelo-llc** [comment]
Wouldn't mind having the ability to specify the parameter name, something like...

Settings > Option > Affiliate > Affiliate Tracking Parameter

The parameter name "tracking" in a URL is definitely problematic with ad blockers.

-----

As for your question, that parameter is referenced in multiple places. Search the _catalog/controller_ folder for `$tracking`

![image](https://github.com/user-attachments/assets/4e3ee144-63fe-411c-89ad-0cc1002c3df1)

**Reply by mavelo-llc** [reply]
https://github.com/search?q=repo%3Aopencart%2Fopencart+path%3A%2F%5Eupload%5C%2Fcatalog%5C%2Fcontroller%5C%2F%2F+%22%24tracking%22&type=code
